homeassistant/components/watchyourlan/sensor.py

```diff
@@ -0,0 +1,149 @@
+"""Support for the WatchYourLAN service."""
+
+from typing import Any
+
+from homeassistant.components.sensor import SensorEntity, SensorEntityDescription
+from homeassistant.config_entries import ConfigEntry
+from homeassistant.core import HomeAssistant
+from homeassistant.helpers.device_registry import CONNECTION_NETWORK_MAC, DeviceInfo
+from homeassistant.helpers.entity_platform import AddEntitiesCallback
+from homeassistant.helpers.update_coordinator import CoordinatorEntity
+
+from .coordinator import WatchYourLANUpdateCoordinator
+
+# Define entity descriptions for each sensor type
+ENTITY_DESCRIPTIONS = [
+    SensorEntityDescription(
+        key="online_status",
+        name="Online Status",
+        icon="mdi:lan-connect",
+    ),
+    SensorEntityDescription(
+        key="ip_address",
+        name="IP Address",
+        icon="mdi:ip-network",
+    ),
+    SensorEntityDescription(
+        key="mac_address",
+        name="MAC Address",
+        icon="mdi:lan",
+    ),
+    SensorEntityDescription(
+        key="iface",
+        name="Network Interface",
+        icon="mdi:ethernet",
+    ),
+]
+
+
+async def async_setup_entry(
+    hass: HomeAssistant, entry: ConfigEntry, async_add_entities: AddEntitiesCallback
+) -> None:
+    """Set up the WatchYourLAN sensors."""
+    coordinator: WatchYourLANUpdateCoordinator = entry.runtime_data
+
+    entities: list[SensorEntity] = [WatchYourLANDeviceCountSensor(coordinator)] + [
+        WatchYourLANGenericSensor(coordinator, device, description)
+        for device in coordinator.data
+        for description in ENTITY_DESCRIPTIONS
+    ]
+
+    async_add_entities(entities)
+
+
+class WatchYourLANGenericSensor(
+    CoordinatorEntity[WatchYourLANUpdateCoordinator], SensorEntity
+):
+    """Generic WatchYourLAN sensor for different data points."""
+
+    def __init__(
+        self,
+        coordinator: WatchYourLANUpdateCoordinator,
+        device: dict[str, Any],
+        description: SensorEntityDescription,
+    ) -> None:
+        """Initialize the sensor."""
+        super().__init__(coordinator)
+        self.device = device
+        self.entity_description = description
+        self._attr_unique_id = f"{self.device.get('ID')}_{description.key}"
```

Comment:
Why don't we use mac address?

Reply:
The schema in WatchYourLAN uses "ID" as the primary key and "mac" doesn't *have* to be unique. Also, if the WatchYourLAN DB gets wiped out, it's possible a previous ID now maps to a different entity within WatchYourLAN or may not exist at all anymore. It seems more inline with how WatchYourLAN works to base the unique_id on the "ID" field instead of the mac addr.

Reply:
> it's possible a previous ID now maps to a different entity within WatchYourLAN or may not exist at all anymore

That kinda sounds like a reason to use the mac address. Yes, maybe the same mac address could appear in the data twice, but they are still used for the same device

Reply:
Or am I missing something here?

Reply:
I still have this question by the way

Reply:
Oh, weird, I thought I committed my change to switch this to the Mac Addr. I think you make a good point here regarding the Mac Addr being the definitive/unique thing, and it makes sense to use it for the device unique ID.